### PR TITLE
Updating Payment Card to be more flexible with accepted props

### PR DESCRIPTION
**Pull request by lukehansell**
#### What does this PR do? (please provide any background)
Allows you to pass extra props into the PaymentCard component.

This is required to replicate existing functionality which responds to onClick events.

Also process fixes:
- change to build the bundled version before release and s3 to make sure s3 gets the latest version.
- ignoring built docs bundle asset.

#### What tests does this PR have?
I've added tests for passing an `onClick` handler and and `id` 

#### How can this be tested?
Run the docs with `npm run docs`
In the browser edit the example for the PaymentCard
Check that your changes are reflected in the rendered DOM

#### What gif best describes how you feel about this work?

---

- [x] I have checked our general [contributing document](https://github.com/holidayextras/culture/blob/master/CONTRIBUTING.md) and the project specific [contributing document](../blob/master/CONTRIBUTING.md) (if present) and I'm happy for this to be reviewed.

#### Reviewers

**Review 1**
- [x] :+1:

**Review 2** \*
- [x] :+1:

**Review 3** _(optional)_
- [ ] :+1:

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.

\*  for HX this review must be completed by an SE, SA or Project Guru

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -33,3 +33,4 @@ docs/npm-debug.log*
 
 # Built files
 dist/components
+docs/public/assets/bundle.js
```
Comment (lukehansell) [inline .gitignore]: built file which shouldn't be committed

File: Gruntfile.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ module.exports = function(grunt) {
 
   gtx.alias('release-docs', ['build', 'docs-build', 'gh-pages']);
 
-  gtx.alias('deploy', ['release', 'aws_s3']);
+  gtx.alias('deploy', ['build', 'release', 'aws_s3']);
```
Comment (lukehansell) [inline Gruntfile.js]: adding 'build' to generate built file for publish to s3

**Comment by hxtomprice** [comment]
:one:

**Comment by shackpank** [comment]
Seems like it could be time for a `1.0` release - we've only just got tripapp and all its dependencies sharing a version, this will re-duplicate it and fixing the duplication pre-1.0 means we'll have to fix it all over again for `0.31`.

**Comment by hxtomprice** [comment]
:+1:

**Comment by jamesmcneil** [comment]
Taking :two:

**Comment by jamesmcneil** [comment]
:+1:

**Pull request merged**